src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -17,10 +18,10 @@
     make_camel_case)
 
 def register_folded_cli_argument(scope, base_name, resource_type, parent_name=None, # pylint: disable=too-many-arguments
-                                 parent_type=None, type_field=None,
+                                 parent_option_flag=None, parent_type=None, type_field=None,
                                  existing_id_flag_value='existingId', new_flag_value='new',
                                  none_flag_value='none', default_value_flag='new',
-                                 **kwargs):
+                                 base_required=True, **kwargs):
```

Comment:
This change is from the original closed PR. Once this conversion is done, this method `register_cli_folder_argument` will be removed entirely.